common/predictive-text/worker/correction/context-tracker.ts

```diff
@@ -351,7 +351,10 @@ namespace correction {
       }
 
       const hasDistribution = transformDistribution && Array.isArray(transformDistribution);
-      const primaryInput = hasDistribution ? transformDistribution[0].sample : null;
+      let primaryInput = hasDistribution ? transformDistribution[0].sample : null;
+      if(primaryInput && primaryInput.insert == "" && primaryInput.deleteLeft == 0 && primaryInput.deleteRight == 0) {
+        primaryInput = null;
```

Comment:
Context checks/matching/updates may occur on empty transforms.  In these cases, we shouldn't add an 'empty' transform as if it corresponded to an actual input.  That scenario (empty transform as input) is what caused the original bug targeted by this PR.

Reply:
I wonder if we could turn this into a function to clarify its purpose: `isEmptyTransform(primaryInput)`?

Reply:
There actually _is_ one, but its accessibility modifiers are inconvenient:

https://github.com/keymanapp/keyman/blob/58a2bda5344a4fa153f772614d644f30b919ec46/common/predictive-text/worker/model-compositor.ts#L40-L42

It's _probably_ best moved into `@keymanapp/models-templates` along with some of the other utility methods, but changing that felt out of scope for this PR.  And is technically a bit of a refactor.

Reply:
Interesting -- that makes it look like `isEmpty` needs work itself to support this PR? Because it doesn't test `deleteRight`...